content/200-concepts/300-more/400-comparisons/04-prisma-and-drizzle.mdx

```diff
@@ -0,0 +1,448 @@
+const posts = await prisma.post.findMany({
+  where: {
+    title: 'Hello World',
+    mode: 'insensitive',
+  },
+})
+```
+
+</block>
+
+<block content="Drizzle">
+
+```ts
+// case sensitive filter
+const posts = await db
+  .select()
+  .from(posts)
+  .where(like(posts.title, 'Hello World'))
+
+// case insensitive filter
+const posts = await db
+  .select()
+  .from(posts)
+  .where(ilike(posts.title, 'Hello World'))
+```
+
+</block>
+
+</ParallelBlocks>
+
+<ParallelBlocks>
+
+<block content="Prisma">
+
+```ts
+const posts = await prisma.post.findMany({
+  where: {
+    title: { contains: 'Hello World' },
+  },
+})
+```
+
+</block>
+
+<block content="Drizzle">
+
+```ts
+const posts = await db
+  .select()
+  .from(posts)
+  .where(ilike(posts.title, '%Hello World%'))
+```
+
+</block>
+
+</ParallelBlocks>
+
+<ParallelBlocks>
+
+<block content="Prisma">
+
+```ts
+const posts = await prisma.post.findMany({
+  where: {
+    title: { startsWith: 'Hello World' },
+  },
+})
+```
+
+</block>
+
+<block content="Drizzle">
+
+```ts
+const posts = await db
+  .select()
+  .from(posts)
+  .where(ilike(posts.title, 'Hello World%'))
+```
+
+</block>
+
+</ParallelBlocks>
+
+<ParallelBlocks>
+
+<block content="Prisma">
+
+```ts
+const posts = await prisma.post.findMany({
+  where: {
+    title: { endsWith: 'Hello World' },
+  },
+})
+```
+
+</block>
+
+<block content="Drizzle">
+
+```ts
+const posts = await db
+  .select()
+  .from(posts)
+  .where(ilike(posts.title, '%Hello World'))
+```
+
+</block>
+
+</ParallelBlocks>
+
+### Pagination
+
+Drizzle only offers limit-offset pagination while Prisma conveniently provides dedicated APIs for both limit-offset but also cursor-based. You can learn more about both approaches in the [Pagination](https://www.prisma.io/docs/concepts/components/prisma-client/pagination) section of the docs.
+
+<ParallelBlocks>
+
+<block content="Prisma">
+
+```ts
+// limit-offset pagination
+const postPage = await prisma.post.findMany({
+  where: {
+    title: 'Hello World',
+  },
+  skip: 6,
+  take: 3,
+})
+// cursor-based pagination
+const postPage = await prisma.post.findMany({
+  where: {
+    title: 'Hello World',
+  },
+  cursor: { id: 7 },
+  take: 3,
+})
+```
+
+</block>
+
+<block content="Drizzle">
+
+```ts
+// limit-offset pagination (cursor-based not currently possible)
+const postPage = await db
+  .select()
+  .from(users)
+  .where(ilike(posts.title, 'Hello World%'))
+  .limit(3)
+  .offset(6)
+```
+
+</block>
+
+</ParallelBlocks>
+
+### Companion APIs
+
+Prisma has many features built into the client that help teams get a better understanding of their data usage. [Metrics](https://www.prisma.io/docs/concepts/components/prisma-client/metrics) and [tracing](https://www.prisma.io/docs/concepts/components/prisma-client/opentelemetry-tracing) are two features that can be enabled at any time and give you per query information. This information can be integrated with external tools so that you can track performance over time.
```

Comment:
```suggestion
Prisma has many features built into the client that help teams better understand their data usage. [Metrics](https://www.prisma.io/docs/concepts/components/prisma-client/metrics) and [tracing](https://www.prisma.io/docs/concepts/components/prisma-client/opentelemetry-tracing) are two features that can be enabled at any time and give you per query information. This information can be integrated with external tools so that you can track performance over time.
```